sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/HttpAuthorization.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.http;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * Represents the value of an HTTP Authorization header.
+ */
+public class HttpAuthorization {
+    private final ClientLogger logger = new ClientLogger(DynamicRequest.class);
+    private final String scheme;
+    private final String parameter;
+
+    /**
+     * Constructs a new HttpAuthorization instance.
+     *
+     * @param scheme Scheme component of an authorization header value.
+     * @param parameter The credentials used for the authorization header value.
+     */
+    public HttpAuthorization(String scheme, String parameter) {
+        if (CoreUtils.isNullOrEmpty(scheme)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("scheme must be a nonempty string."));
+        }
+        if (CoreUtils.isNullOrEmpty(parameter)) {
+            throw logger.logExceptionAsError(new NullPointerException("parameter must be a nonempty string."));
+        }
```

Comment:
One is throwing NPE and the other is throwing IllegalArgumentException. If the input param is null throw NPE and if it's empty throw IllegalArgumentException. See [Java guidelines](https://azure.github.io/azure-sdk/java_introduction.html#exceptions)